api/v1beta1/openstackdataplanenodeset_webhook.go

```diff
@@ -198,9 +234,63 @@ func (r *OpenStackDataPlaneNodeSetSpec) ValidateUpdate(oldSpec *OpenStackDataPla
 		}
 	}
 
+	// Compare TLS settings of control plane and data plane
+	// if control plane name is specified attempt to retrieve it
+	// otherwise get any control plane in the namespace
+	controlPlane := openstackv1.OpenStackControlPlane{}
+	if len(r.ControlPlane) > 0 {
+		controlPlaneName := types.NamespacedName{
+			Name:      r.ControlPlane,
+			Namespace: namespace,
+		}
+		err := webhookClient.Get(context.TODO(), controlPlaneName, &controlPlane)
+		if err != nil {
+			errors = append(errors, field.InternalError(field.NewPath("spec.tlsEnabled"), err))
+		}
+
+	} else {
+		controlPlanes := openstackv1.OpenStackControlPlaneList{}
+		opts := client.ListOptions{
+			Namespace: namespace,
+		}
+
+		_ = webhookClient.List(context.TODO(), &controlPlanes, &opts)
+
+		// Verify TLS status of control plane only if there is a single one
+		// report error if there are multiple, or proceed if there are none
+		if len(controlPlanes.Items) > 1 {
+			errors = append(errors, field.InternalError(
+				field.NewPath("spec.tlsEnabled"),
+				fmt.Errorf("multiple control planes found in the namespace %v", namespace)))
+		} else if len(controlPlanes.Items) == 1 {
+			controlPlane = controlPlanes.Items[0]
+			err := r.TLSMatch(controlPlane)
+			if err != nil {
+				errors = append(errors, err)
+			}
+		}
+	}
+
 	return errors
 }
 
+// Do TLS flags match in control plane ingress, pods and data plane
+func (r *OpenStackDataPlaneNodeSetSpec) TLSMatch(controlPlane openstackv1.OpenStackControlPlane) *field.Error {
+
+	if controlPlane.Spec.TLS.Ingress.Enabled != r.TLSEnabled || controlPlane.Spec.TLS.PodLevel.Enabled != r.TLSEnabled {
+
+		return field.Forbidden(
+			field.NewPath("spec.tlsEnabled"),
+			fmt.Sprintf(
+				"TLS settings on Data Plane node set and Control Plane %s do not match, Node set: %t Control Plane Ingress: %t Control Plane PodLevel: %t",
+				controlPlane.Name,
+				r.TLSEnabled,
+				controlPlane.Spec.TLS.Ingress.Enabled,
+				controlPlane.Spec.TLS.PodLevel.Enabled))
+	}
+	return nil
+}
```

Comment:
I wonder if this should just be a validation.  Or whether we should make this a mutating webhook and match the setting in the control plane.  Thoughts?

Reply:
Not relevant anymore